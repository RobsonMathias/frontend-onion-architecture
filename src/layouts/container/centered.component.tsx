import React from 'react';
import { Grid, Container, Box } from '@mui/material';

type CenteredComponentProps = React.HTMLAttributes<HTMLDivElement> & {
  children: NonNullable<React.ReactNode>;
  maxWidth?: 'xs' | 'sm' | 'md' | 'lg' | 'xl' | false;
};

export function CenteredComponent(props: CenteredComponentProps) {
  const { children, maxWidth = 'xs', ...isProps } = props;

  return (
    <Box py={10}>
      <Grid {...isProps} container alignItems="center">
        <Container maxWidth={maxWidth}>{children}</Container>
      </Grid>
    </Box>
  );
}
