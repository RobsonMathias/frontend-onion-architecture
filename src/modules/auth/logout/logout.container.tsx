import React from 'react';

import { useAuth } from '../hooks';
import { Button } from '@mui/material';
import { BiLogOut } from 'react-icons/bi';

export function LogoutContainer() {
  const { logout } = useAuth();

  function handleClick() {
    logout();
  }

  return (
    <Button
      onClick={handleClick}
      data-testid="logout"
      type="button"
      variant="outlined"
      color="secondary"
      endIcon={<BiLogOut />}
    >
      Sair
    </Button>
  );
}
