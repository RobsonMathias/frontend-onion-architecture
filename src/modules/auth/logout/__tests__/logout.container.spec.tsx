/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';

import { act, fireEvent, render } from '@testing-library/react';
import React from 'react';

const mockLogout = jest.fn();
const mockUseAuth = jest.fn(() => ({ logout: jest.fn(() => mockLogout()) }));
jest.mock('../../hooks', () => ({ useAuth: jest.fn(() => mockUseAuth()) }));

import { LogoutContainer } from '../logout.container';

describe('LogoutContainer', () => {
  it('should render successfully', () => {
    const { container } = render(<LogoutContainer />);
    expect(container).toMatchSnapshot();
  });

  it('should call logout successfully', async () => {
    const { getByTestId } = render(<LogoutContainer />);
    await act(async () => {
      fireEvent.click(getByTestId('logout'));
    });
    expect(mockLogout).toHaveBeenCalled();
  });
});
