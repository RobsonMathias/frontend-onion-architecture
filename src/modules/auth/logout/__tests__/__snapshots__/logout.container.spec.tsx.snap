// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutContainer should render successfully 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-wnpvqa-MuiButtonBase-root-MuiButton-root"
    data-testid="logout"
    tabindex="0"
    type="button"
  >
    Sair
    <span
      class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16 13v-2H7V8l-5 4 5 4v-3z"
        />
        <path
          d="M20 3h-9c-1.103 0-2 .897-2 2v4h2V5h9v14h-9v-4H9v4c0 1.103.897 2 2 2h9c1.103 0 2-.897 2-2V5c0-1.103-.897-2-2-2z"
        />
      </svg>
    </span>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
